x-pack/plugins/security/public/session/session_expired.ts

```diff
@@ -4,26 +4,28 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { HttpSetup } from 'src/core/public';
-
 export interface ISessionExpired {
   logout(): void;
 }
 
+const getNextParameter = () => {
+  const { location } = window;
+  const next = encodeURIComponent(`${location.pathname}${location.search}${location.hash}`);
+  return `&next=${next}`;
+};
+
+const getProviderParameter = (tenant: string) => {
+  const key = `${tenant}/session_provider`;
+  const providerName = sessionStorage.getItem(key);
+  return providerName ? `&provider=${encodeURIComponent(providerName)}` : '';
+};
+
 export class SessionExpired {
-  constructor(private basePath: HttpSetup['basePath'], private tenant: string) {}
+  constructor(private logoutUrl: string, private tenant: string) {}
 
   logout() {
-    const next = this.basePath.remove(
-      `${window.location.pathname}${window.location.search}${window.location.hash}`
-    );
-    const key = `${this.tenant}/session_provider`;
-    const providerName = sessionStorage.getItem(key);
-    const provider = providerName ? `&provider=${encodeURIComponent(providerName)}` : '';
-    window.location.assign(
-      this.basePath.prepend(
-        `/logout?next=${encodeURIComponent(next)}&msg=SESSION_EXPIRED${provider}`
-      )
-    );
+    const next = getNextParameter();
+    const provider = getProviderParameter(this.tenant);
+    window.location.assign(`${this.logoutUrl}?msg=SESSION_EXPIRED${next}${provider}`);
```

Comment:
I broke out the logic of constructing the query parameters so this is easier to understand.